src/rust/src/lib.rs

```diff
@@ -91,6 +96,11 @@ fn _rust(py: pyo3::Python<'_>, m: &pyo3::types::PyModule) -> pyo3::PyResult<()>
     crate::x509::ocsp_resp::add_to_module(ocsp_mod)?;
     m.add_submodule(ocsp_mod)?;
 
+    let openssl_mod = unsafe {
+        let ptr = PyInit__openssl();
+        pyo3::types::PyModule::from_owned_ptr(py, ptr)
+    };
+    m.add_submodule(openssl_mod)?;
```

Comment:
Instead of making `openssl` a sub-module of `_rust`, can we instead make the heirarchy the same as it is now. That is:

- Change `src/rust/` to be `cryptography.hazmat.bindings`
- `m.add_submodule(openssl_mod)`
- `m.add_submodule(rust_mod)`, where `rust_mod` is a new module which has all the current sub-modules as sub-modules of _it_.

Reply:
I can do this after building and testing on CI works. But I'm not sure how to deal with these Python files in  https://github.com/pyca/cryptography/tree/main/src/cryptography/hazmat/bindings/openssl

Reply:
```diff
diff --git a/setup.cfg b/setup.cfg
index 23a6a55b..45ad765f 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -43,7 +43,7 @@ zip_safe = False
 package_dir =
     =src
 packages = find:
-ext_package = cryptography.hazmat.bindings
+ext_package = cryptography.hazmat
 # `install_requires` must be kept in sync with `pyproject.toml`
 install_requires =
     cffi >=1.12
diff --git a/setup.py b/setup.py
index 8f781a6a..8a6e6433 100644
--- a/setup.py
+++ b/setup.py
@@ -39,7 +39,7 @@ try:
     setup(
         rust_extensions=[
             RustExtension(
-                "_rust",
+                "bindings",
                 "src/rust/Cargo.toml",
                 py_limited_api=True,
                 # Enable abi3 mode if we're not using PyPy.
diff --git a/src/cryptography/hazmat/bindings/openssl/binding.py b/src/cryptography/hazmat/bindings/openssl/binding.py
index 5ad08189..21daadff 100644
--- a/src/cryptography/hazmat/bindings/openssl/binding.py
+++ b/src/cryptography/hazmat/bindings/openssl/binding.py
@@ -12,11 +12,11 @@ import warnings
 import cryptography
 from cryptography import utils
 from cryptography.exceptions import InternalError
-from cryptography.hazmat.bindings._rust import _openssl
+from cryptography.hazmat.bindings._openssl import ffi, lib
 from cryptography.hazmat.bindings.openssl._conditional import CONDITIONAL_NAMES

-ffi = _openssl.ffi
-lib = _openssl.lib
+# ffi = _openssl.ffi
+# lib = _openssl.lib
 _OpenSSLErrorWithText = collections.namedtuple(
     "_OpenSSLErrorWithText", ["code", "lib", "reason", "reason_text"]
 )
diff --git a/src/rust/src/lib.rs b/src/rust/src/lib.rs
index 0b7c98f1..c91aaa7a 100644
--- a/src/rust/src/lib.rs
+++ b/src/rust/src/lib.rs
@@ -100,11 +100,18 @@ fn _rust(py: pyo3::Python<'_>, m: &pyo3::types::PyModule) -> pyo3::PyResult<()>
     crate::x509::ocsp_resp::add_to_module(ocsp_mod)?;
     m.add_submodule(ocsp_mod)?;

+    Ok(())
+}
+
+#[pyo3::prelude::pymodule]
+fn bindings(py: pyo3::Python<'_>, m: &pyo3::types::PyModule) -> pyo3::PyResult<()> {
     let openssl_mod = unsafe {
         let ptr = PyInit__openssl();
         pyo3::types::PyModule::from_owned_ptr(py, ptr)
     };
     m.add_submodule(openssl_mod)?;
+
+    m.add_wrapped(pyo3::wrap_pymodule!(_rust))?;
     Ok(())
 }

diff --git a/tests/hazmat/backends/test_openssl_memleak.py b/tests/hazmat/backends/test_openssl_memleak.py
index d1d672a4..3bae0243 100644
--- a/tests/hazmat/backends/test_openssl_memleak.py
+++ b/tests/hazmat/backends/test_openssl_memleak.py
@@ -24,10 +24,10 @@ def main(argv):

     import cffi

-    from cryptography.hazmat.bindings._rust import _openssl
+    from cryptography.hazmat.bindings._openssl import ffi, lib

-    ffi = _openssl.ffi
-    lib = _openssl.lib
+    # ffi = _openssl.ffi
+    # lib = _openssl.lib

     heap = {}
```

This does not work 

```python
ImportError while loading conftest '/Users/messense/Projects/cryptography/tests/conftest.py'.
tests/conftest.py:8: in <module>
    from cryptography.hazmat.backends.openssl import backend as openssl_backend
src/cryptography/hazmat/backends/openssl/__init__.py:6: in <module>
    from cryptography.hazmat.backends.openssl.backend import backend
src/cryptography/hazmat/backends/openssl/backend.py:13: in <module>
    from cryptography import utils, x509
src/cryptography/x509/__init__.py:6: in <module>
    from cryptography.x509 import certificate_transparency
src/cryptography/x509/certificate_transparency.py:10: in <module>
    from cryptography.hazmat.bindings._rust import x509 as rust_x509
E   ImportError: cannot import name 'x509' from 'cryptography.hazmat.bindings._rust' (unknown location)
```

Reply:
@alex what do you think?

Reply:
I think let's not worry about including it in this PR, it'd be cool to have the module names stay teh same, but not worth the headache!